### PR TITLE
[Simplifying Deposits UI] Update the deposit status chip to use our new WC Pay Pill

### LINKED ISSUE #5892: [Simplifying Deposits UI] Update the DepositStatusChip to make use of updated WCPay Pill
### Description

In https://github.com/Automattic/woocommerce-payments/pull/5890 we have updated the WCPay Pill component to use `@woocommerce/components/pill` instead of our own styled span tags.

In this issue we will updating the [`<DepositStatusChip />`](https://github.com/Automattic/woocommerce-payments/blob/53938f33b37307f469ec9a947de9891115f8b4e9/client/components/deposit-status-chip/index.tsx) to use our own Pill component and also rename it to `<DepositStatusPill />`

This component will take a deposit status as a prop and map it to one of the new Pill types (primary, success, alert, danger & light) and render the different deposit status pills as per the latest designs:

image

Usage:

```
<DepositStatusPill status="estimated" />
<DepositStatusPill status="paid" />
<DepositStatusPill status="in_transit" />
```

Output:

image

### PR DESCRIPTION
**Pull request by mattallan**
> **Warning** - This PR is based off #5890 which should be merged before this branch

Fixes #5892

#### Changes proposed in this Pull Request

In https://github.com/Automattic/woocommerce-payments/pull/5890 we updated our WCPay pill component to use of the [WC Core `<Pill>`](https://woocommerce.github.io/woocommerce-admin/components/storybook/index.html?path=/story/woocommerce-admin-components-pill--default) component, in this PR I'm updating our `<DepositStatusChip />` to use our new Pill component and map the given deposit status to the correct pill type for styling and correct status label.

The changes coming in this PR include:
 - Rename the component to `DepositStatusPill`.
 - Given a deposit status, map the status to the Pill type and render the pill label
 - Update the existing `components/deposits-overview/next-deposit` component to use the new deposit status pill.

**Usage**

```
<DepositStatusPill status="estimated" />
<DepositStatusPill status="pending" />
<DepositStatusPill status="in_transit" />
<DepositStatusPill status="paid" />
<DepositStatusPill status="failed" />
<DepositStatusPill status="canceled" />
```

**Output**

image

#### Testing instructions

1. Checkout this branch and run `npm run build:client`
2. Set `_wcpay_feature_simplify_deposits_ui` to `1` in wp-options table
3. Go to the Payments > Overview page
4. Confirm the Next Deposit table has an estimated deposits status pill styled correctly
5. Since we don't have any other deposit statuses using this pill you can use the above code to render all of the deposit pills

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5946` or branch name `issue/5892-update-deposit-status-chip` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=issue/5892-update-deposit-status-chip) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8ca616e8b6bf8756f064563339a26f4d3ee7bf9e
- Build time: 2023-04-11 01:17:19 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/deposit-status-pill/index.tsx
Comment (james-allan) [inline client/components/deposit-status-pill/index.tsx:34]: As we spoke about IRL, this `status as keyof typeof displayStatus` could be improved if the deposit status was a defined type. In the interest of moving this forward, I'm going to approve this PR and address the much wider type refactor as part of separate PR.

**Review by james-allan — APPROVED** [review]

File: client/components/deposit-status-pill/index.tsx
Comment (james-allan) [reply client/components/deposit-status-pill/index.tsx:34]: New issue: https://github.com/Automattic/woocommerce-payments/issues/5987

**Review by james-allan — APPROVED** [review]

**Comment by mattallan** [comment]
I've just pushed up [c1b6c63](https://github.com/Automattic/woocommerce-payments/pull/5946/commits/c1b6c632c0819d58e99be2973800a1badafcf93d) which updates the Recent Deposits table to also use the new deposits pill:

image

---

This is a very minor change after it was approved so I'll wait for the checks to pass, then merge this

**Pull request merged**
